sdk/identity/identity/test/internal/node/workloadIdentityCredentialIdentityBinding.spec.ts

```diff
@@ -0,0 +1,904 @@
+      await new Promise((resolve) => setTimeout(resolve, 10));
+
+      // Fourth call should succeed again with the new valid certificate
+      const tlsSettings3 = await (credential as any).getTlsSettings();
+      assert.isDefined(tlsSettings3);
+      assert.equal(tlsSettings3.ca, getTestCertificateContent());
+
+      // Should be a new object reference since cache was invalidated
+      assert.equal(tlsSettings3.ca, getTestCertificateContent());
+      await fs.unlink(tempCaFile);
+      await fs.rmdir(tempDir);
+    });
+
+    it("should handle empty CA file during rotation", async function () {
+      const tempDir = await fs.mkdtemp(path.join(os.tmpdir(), "cert-test-"));
+      const tempCaFile = path.join(tempDir, "ca.pem");
+
+      await fs.copyFile(TEST_CERT_PATH, tempCaFile);
+
+      vi.stubEnv("AZURE_KUBERNETES_TOKEN_PROXY", "https://test-proxy.example.com");
+      vi.stubEnv("AZURE_KUBERNETES_CA_FILE", tempCaFile);
+
+      const credential = new WorkloadIdentityCredential({
+        tenantId,
+        clientId,
+        tokenFilePath,
+        enableAzureKubernetesTokenProxy: true,
+      });
+
+      // First call should succeed and cache the TLS settings
+      const tlsSettings1 = await (credential as any).getTlsSettings();
+      assert.isDefined(tlsSettings1);
+      assert.equal(tlsSettings1.ca, getTestCertificateContent());
+
+      // Simulate CA rotation by emptying the file
+      await fs.writeFile(tempCaFile, "", "utf8");
+      await new Promise((resolve) => setTimeout(resolve, 10));
+
+      // Call with empty file should return cached settings
+      const tlsSettings2 = await (credential as any).getTlsSettings();
+      assert.strictEqual(tlsSettings1, tlsSettings2);
+
+      const credential2 = new WorkloadIdentityCredential({
+        tenantId,
+        clientId,
+        tokenFilePath,
+        enableAzureKubernetesTokenProxy: true,
+      });
+
+      let rejectionError: Error | undefined;
+      try {
+        await (credential2 as any).getTlsSettings();
+      } catch (error) {
+        rejectionError = error as Error;
+      }
+      assert.isDefined(rejectionError);
+      assert.match(rejectionError!.message, /CA certificate file is empty/);
+      await fs.unlink(tempCaFile);
+      await fs.rmdir(tempDir);
+    });
+  });
+
+  describe("URL Rewriting", function () {
+    let pipelineModule: any;
+    let createDefaultHttpClientSpy: any;
+
+    beforeEach(async function () {
+      pipelineModule = await import("@azure/core-rest-pipeline");
+      createDefaultHttpClientSpy = vi.spyOn(pipelineModule, "createDefaultHttpClient");
+    });
+
+    afterEach(function () {
+      vi.unstubAllEnvs();
+      vi.restoreAllMocks();
+    });
+
+    const testCases = [
+      {
+        name: "proxy url with / path; request path has no leading slash",
+        proxyUrl: "https://proxy.example.com/",
+        requestUrl: "https://orig.example.com/login?a=1&b=2",
+        wantScheme: "https",
```

Comment:
you could just hardcode this in the assertion because no case uses a different scheme